### DISCUSSION TITLE
ai deck tutorial

### DISCUSSION THREAD

**Discussion by Rwik2000** [bitcraze/discussions#986] [Q&A] [answered]
I am trying to install ai deck on the crazyflie. I went through the documentation: [https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#gap8-bootloader](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#gap8-bootloader) , is the Jlink necessary to update the bootloader the first time?

I found this link [https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/infrastructure/flashing/](https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/infrastructure/flashing/) and was trying to flash using the cfloader, but the process got stuck at 0% and was not going ahead.

Can you confirm if the Jlink is required or is there anyway we could do using the radio?

**Comment by knmcguire** [comment]
Hi! 

When you were going through the tutorial, were you able to flash the deck lik in this section? https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#update-crazyflie-and-aideck-firmware

And flash the prebuild wifi example as shown here: https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#flash-wifi-example ?

If not, please first do all of the getting started tutorial first before moving to the examples. If you still have the flashing issue there, then we can debug your issue from there.

**Reply by Rwik2000** [reply]
Yes, I flashed the deck link. But I could NOT see `ESP32: I (910) SYS: Initialized` in the cfclient window.

As for the second case, the process stays at 0% and does not increase.

Will I need a Jlink at any point? or the radio would suffice?

**Reply by knmcguire** [reply]
Ah oke, if you don't see that, then you should try flashing it according to those instructions again. As the notes says in the tutorail
```
NOTE: If you don’t see ESP32: I (910) SYS: Initialized, try go through the steps again.
```
Mind that flashing the deck will take a few minutes, so don't disconnect or anything. If it doesn't take that long or there is no clear progression of the flashing bar of the decks, that means something has failed too. Make sure that there is no other deck connected and that only the aideck is connected.

The next phases won't do anything anyway until the ESP is properly initialize so try it again for a couple of times.

**Reply by Rwik2000** [reply]
So I always get this 
![image](https://github.com/bitcraze/discussions/assets/52771866/6b4c2186-c809-4f62-9ae0-295768be82fc)

But, I do not see `ESP32: I (910) SYS: Initialized`

These are the logs in terminal : 

```
INFO:cflib.crazyflie:Callback->Connection initialized[radio://0/80/2M/E7E7E7E702]
INFO:cflib.crazyflie:We are connected[radio://0/80/2M/E7E7E7E702], request connection setup
INFO:cflib.crazyflie.platformservice:Request _request_protocol_version()
INFO:cflib.crazyflie:Callback->Connected to [radio://0/80/2M/E7E7E7E702]
INFO:cflib.crazyflie.platformservice:_crt_service_callback
INFO:cflib.crazyflie.platformservice:Request protocol version
INFO:cflib.crazyflie.platformservice:_platform_callback
INFO:cflib.crazyflie.platformservice:Protocol version (platform): 5
WARNING:cflib.crazyflie.toccache:Could not save cache, no writable directory
INFO:cflib.crazyflie:Log TOC finished updating
INFO:cflib.crazyflie.mem:5 memories found
INFO:cflib.crazyflie:Memories finished updating
WARNING:cflib.crazyflie.toccache:Could not save cache, no writable directory
INFO:cflib.crazyflie:Param TOC finished updating
INFO:cflib.crazyflie:Callback->Connection setup finished [radio://0/80/2M/E7E7E7E702]
Deck bcAI:esp, reset to bootloader
INFO:cflib.crazyflie:All parameters updated
INFO:cflib.crazyflie:Callback->Connection completed [radio://0/80/2M/E7E7E7E702]
INFO:cflib.crazyflie:Closing link
INFO:cflib.crazyflie:Callback->Disconnected from [radio://0/80/2M/E7E7E7E702]
```

After the bootloading step, here is output of the console tab once I reconnect : 

```
SYS: ----------------------------
SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:8c606d1f8f89 (2022.05) CLEAN
SYS: I am 0x203235324133501900430032 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 1 deck(s) found
DECK_CORE: Calling INIT on driver bcAI for deck 0
AIDECK-ROUTER: Not setting up WiFi
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2022.0
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
DECK_CORE: Deck 0 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 16616 bytes
STAB: Ready to fly.
```

**Reply by Rwik2000** [reply, marked as answer]
I just realised from one of your other comments in a separate thread that 

```
pip3 uninstall cflib cfclient
pip3 install cfclient
```

works!
